### ISSUE TITLE
Health system

### ISSUE THREAD

**Issue by rm-code** [rm-code/On-The-Roadside#72]
Some more brain storming stuff ... 
![human](https://cloud.githubusercontent.com/assets/11627131/20863379/d159d6d4-b9c8-11e6-95f0-ae594d78c5de.png)

The bodies of all creatures in the game will be modelled as graph like the on above. These graphs will feature entry nodes, bones and organs.

## Nodes
- Entry nodes (pink) are used to represent the _outside_ of the body. These are points which can be targeted and hit directly. E.g.: The player attacks an arm or the head of a creature.
- The damage of the attack would then be conferred to the inside of the body where it can hit bones (grey) and / or organs (green).

## Damage propagation between nodes
Damage to a creature's body will be affected by the [Damage Type](https://github.com/rm-code/On-The-Roadside/issues/74) of the used weapon / ammunition.

In reality a swing with a baseball bat could hit both the arm and the chest of the defender. Therefore it would make sense to allow damage propagation between entry nodes for certain types of damage:
- Bludgeoning damage would have a high chance to damage adjacent nodes and bones, but not internal organs.
- Piercing damage would directly travel from the entry node to an end node and have a high chance to hit the inner nodes.
- Slashing damage would be able to propagate its damage to an adjacent node, but have a lower chance to hit vital (inner) organs.

## Effects based on nodes
Nodes could be marked with certain effects which would then be removed upon their destruction.

**Comment by rm-code** [comment]
I'm currently thinking about a simplified approach: 
![screenshot 2016-11-15 23 25 24](https://cloud.githubusercontent.com/assets/11627131/20326847/d1ff32ce-ab8a-11e6-8ce2-93f0f7a24d0f.png)

Regardless of which system I'll use I'd probably have to simulate the force of an attack or some other way to have realistic outcomes.

With the simpler approach an attack could only directly hit the (orange) entry nodes and the damage would be directed to the connected nodes.  The purple end nodes represent organs and are basically the final receivers of the damage. Green nodes would have a special role. They act as "containers" for the purple nodes they are connected to and can direct the damage they receive to one or more (!) of their "children". 

> E.g. if an attack hits the head the damage would be dealt to the skull which could crack, but still protect the brain from taking damage.

Compared to the approach in the first post you wouldn't have damage propagation between end nodes anymore, but maybe this could still be applied on the entry node layer.

**Comment by rm-code** [comment]
Some more brain storming stuff ... 
![body](https://cloud.githubusercontent.com/assets/11627131/20462975/a3c5df32-af29-11e6-85e8-b4728eb19fef.png)

Pink nodes are entry nodes which can be directly hit by an attack. Grey nodes are bones which have DR against certain attacks. Green nodes represent organs. If a vital organ is destroyed the creature dies. 

- Piercing damage would directly travel from the entry node to an end node and have a high chance to hit the inner nodes.
- Slashing damage would be able to propagate its damage to an adjacent node, but have a lower chance to damage and therefore a lower chance to hit vital organs
- Bludgeoning damage would have a high chance to damage adjacent nodes and bones, but not internal organs

**Comment by rm-code** [comment]
Original idea by harakka.

![](https://dl.dropboxusercontent.com/u/85130/body.png)

slashing attack to body_lower
body_lower -> thigh_l 20%
body_lower -> body_middle 30%
body_lower -> intestine 70%
body_lower -> pelvicbone 30%
(when you see _r imagine a _l mirror image)
blunt attack to body_lower: works basically the same way since there's no skeletal structure protecting the organs, but you can add a modifier to blunt damage that lowers the probability or damage of propagation, so soft tissues and shielding skeletal structures get to protect vital organs
case in point, more complicated examples:
slashing attack to body_upper
body_upper -> neck 30%
body_upper -> arm_upper_l 20%
body_upper -> spine_upper 10%
body_upper -> ribcage 70%

yay all cool organs are inside the ribcage so slashing attacks don't damage vitals as long as ribcage is intact

blunt attack to body_upper
body_upper -> neck 30%
body_upper -> arm_upper_l 20%
body_upper -> spine_upper 10%
body_upper -> ribcage 70%
ribcage -> heart 30%
ribcage -> lungs 70%
however since ribcage is a skeletal container, we should only propagate very minimal amount of damage from ribcage->heart and lungs on blunt attacks
and, if ribcage was already broken at this point, damage would go straight through (imagine someone hitting you in the chest with a baseball bat when you have multiple broken or shattered ribs, there's probably a decent chance of a rib piercing a lung)

piercing attack to body_upper
body_upper -> neck 30%
body_upper -> arm_upper_l 20%
body_upper -> spine_upper 10%
body_upper -> ribcage 70%
ribcage -> heart 30%
ribcage -> lungs 70%
the point of piercing is that it pierces so for example hit that gets through soft tissue to ribcage has 70% chance of hitting the lung and 30% heart, if you want to make it a bit less lethal you could only propagate into single node inside a container. i considered piercing as sort of a default case and the edge weights are based on that, but you could modify damage so that damage to soft tissue and skeletal bits is reduced

**Comment by rm-code** [comment]
Quick brain storming for bleeding:
- Amount of bleeding per organ / node is reduced each turn
- Nodes have custom bleed factors which determine the amount of lost blood per turn
- Damage can affect the bleed factors (high damage -> more bleeding)
    - slashing and stabbing should have a high bleeding factor
    - bludgeoning should reduce the bleeding factor

**Comment by rm-code** [comment]
After working on this for what feels like ages I have found a decent approach. I simplified the body model, but also managed to link equipment slots to different body parts.

![screenshot 2017-01-15 02 19 42](https://cloud.githubusercontent.com/assets/11627131/21959533/449413f4-daca-11e6-8760-525573eb64b4.png)

I'm closing this ticket now since it is quite broad. New tickets will be opened for more specific features based on the base body model I created.
